### DISCUSSION TITLE
Use table to search and select row for edit

### DISCUSSION THREAD

**Discussion by lozdown** [sqlpage/SQLPage#151] [Q&A] [answered]
I'm new to SQLPage. How to select an item from a table and pass it to an edit form?

**Comment by lovasoa** [comment, marked as answer]
Hello and welcome to SQLPage !

The easiest way to achieve what you want to do is to add an "edit" column to your table with a link to your edition page. Like so:

```sql
select 'table' as component, 'action' as markdown;
select *, format('[Edit](edit.sql?id=%s)', id) as action from users;
```

Then, in `edit.sql`, you can have your form:

```sql
select 'form' as component;
select 'text' as type, 'Username' as name, username as value from users where id = $id;

update users set username = :Username where id = $id and :Username is not null;
```

See the [simple website example](https://github.com/lovasoa/SQLpage/tree/main/examples/simple-website-example)

| `table.sql` | `edit.sql` | 
| --- | --- |
| ![image](https://github.com/lovasoa/SQLpage/assets/552629/5b3af428-0bae-49ee-9fa3-bb2cb2b22226) | ![image](https://github.com/lovasoa/SQLpage/assets/552629/4e4e9577-8b5e-4786-a8b7-7cc328b55908) |

**Reply by lozdown** [reply]
Very helpful - thanks
